ldes-server-retention/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/services/ViewDeletedEventHandler.java

```diff
@@ -0,0 +1,24 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.retention.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.view.valueobject.ViewDeletedEvent;
+import be.vlaanderen.informatievlaanderen.ldes.server.retention.repositories.MemberPropertiesRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.retention.services.retentionpolicy.execution.MemberRemover;
+import org.springframework.context.event.EventListener;
+
+public class ViewDeletedEventHandler {
+	private final MemberPropertiesRepository memberPropertiesRepository;
+	private final MemberRemover memberRemover;
+
+	public ViewDeletedEventHandler(MemberPropertiesRepository memberPropertiesRepository, MemberRemover memberRemover) {
+		this.memberPropertiesRepository = memberPropertiesRepository;
+		this.memberRemover = memberRemover;
+	}
+
+	// call from retentionPolicyCollectionImpl.handleViewDeletedEvent?
+	@EventListener
+	public void handleViewDeletedEvent(ViewDeletedEvent event) {
+		String viewName = event.getViewName().asString();
+		memberPropertiesRepository.getMemberPropertiesWithViewReference(viewName)
+				.forEach(memberProperties -> memberRemover.removeMemberFromView(memberProperties, viewName));
```

Comment:
You can also implement a different repo method which is deletes all members of a view.